File: src/main/java/nl/crosscode/x509certbuddy/models/CertModel.java
package nl.crosscode.x509certbuddy.models;

import java.util.Base64;

import java.security.cert.CertificateEncodingException;
import java.security.cert.X509Certificate;
import java.util.Date;

public class CertModel {
    private final String der;
    private final String serial;
    private final String subject;
    private final String issuer;
    private final Date notBefore;
    private final Date notAfter;
    public CertModel(X509Certificate cert) throws CertificateEncodingException {
        der = Base64.getEncoder().encodeToString(cert.getEncoded());
        serial = cert.getSerialNumber().toString(16);
        subject = cert.getSubjectDN().getName();
        issuer = cert.getIssuerDN().getName();
        notBefore = cert.getNotBefore();
        notAfter = cert.getNotAfter();
    }

    public String getDer() {
        return der;
    }

    public String getSerial() {
        return serial;
    }

    public String getSubject() {
        return subject;
    }

    public String getIssuer() {
        return issuer;
    }

    public Date getNotBefore() {
        return notBefore;
    }

    public Date getNotAfter() {
        return notAfter;
    }

}

File: src/main/kotlin/nl/crosscode/x509certbuddy/models/CertModel.kt
package nl.crosscode.x509certbuddy.models

import java.security.cert.CertificateEncodingException
import java.security.cert.X509Certificate
import java.util.*

class CertModel(cert: X509Certificate) {
    private val der: String
    private val serial: String
    private val subject: String
    private val issuer: String
    private val notBefore: Date
    private val notAfter: Date

    init {
        val der = try {
            Base64.getEncoder().encodeToString(cert.encoded)
        } catch (e: CertificateEncodingException) {
            ""
        }
        this.der = der
        serial = cert.serialNumber.toString(16)
        subject = cert.subjectX500Principal.name
        issuer = cert.issuerX500Principal.name
        notBefore = cert.notBefore
        notAfter = cert.notAfter
    }
}
